src/diffusers/models/unets/unet_3d_condition.py

```diff
@@ -132,13 +132,19 @@ def __init__(
                 "At the moment it is not possible to define the number of attention heads via `num_attention_heads` because of a naming issue as described in https://github.com/huggingface/diffusers/issues/2011#issuecomment-1547958131. Passing `num_attention_heads` will only be supported in diffusers v0.19."
             )
 
+        if isinstance(attention_head_dim, int):
+            num_attention_heads = [out_channels // attention_head_dim for out_channels in block_out_channels]
```

Comment:
is that always correct? Does `out_channels` always represent the `hidden_dim` of the attention layer?

Reply:
For the 3D UNets this is safe. There are a limited number of blocks used with this model `CrossAttnDownBlock3D`, CrossAttnUpBlock3D` and `UNetMidBlock3DCrossAttn` and they all configure `num_attention_heads` based on the out_channels.